internal/users/add.go

```diff
@@ -0,0 +1,77 @@
+// Copyright © 2022 Equinix Metal Developers <support@equinixmetal.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package users
+
+import (
+	"strings"
+
+	"github.com/packethost/packngo"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func (c *Client) Add() *cobra.Command {
+	var email string
+	var roles []string
+	var projectIDs []string
+	var organizationID string
+
+	// addUserCmd represents the addUser command
+	addUserCmd := &cobra.Command{
+		Use:   `add --email <email> --roles <roles> [--organization-id <organization_id>] [--project-id <project_id>]`,
+		Short: "Adds a user to an organization or project",
+		Long:  "Adds a user, by email, to the organization or project specified by the --organization-id or --project-id flag. The user will be assigned the roles specified by the --roles flag.",
+		Example: `  # Adds a user to a project with admin role:
+  metal user add --email user@example.org --roles admin --project-id 3b0795ba-fd0b-4a9e-83a7-063e5e12409d
+`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			getOpts := c.Servicer.ListOptions([]string{"invited_by", "invitable", "projects"}, nil)
+			createRequest := &packngo.InvitationCreateRequest{
+				Invitee:        email,
+				Roles:          roles,
+				ProjectIDs:     projectIDs,
+				OrganizationID: organizationID,
+			}
+			invitation, _, err := c.InvitationService.Create(createRequest, getOpts)
```

Comment:
I'm having trouble identifying why this API call (with a program id, role, organization id, and invitee) is resulting in:
```
{
 "errors": [
  "Roles contains an invalid role limited_collaborator"
 ]
}
```

Making the same request through the web portal (and investigating the network panel) reveals that the requests are identical.  The differences that I feel need to be further scrutinized for this PR, are the HTTP headers.

Reply:
`project_ids` was supposed to be `projects_ids` 🤦